src/analyze_logs.py

```diff
@@ -132,7 +132,11 @@ def log_file_p(file_name):
         files = [os.path.join(log_directory, f)
                  for f in sorted(args.filenames)]
     if args.clear:
-        shutil.rmtree(os.path.join(log_directory, 'log_analyzer_cache'))
+        try:
+            shutil.rmtree(os.path.join(log_directory, 'log_analyzer_cache'))
+        except IOError:
+            print("Error trying to clear the directory. Has the clear been called already?")
```

Comment:
I'd suggested printing the error message from the exception here. There may be other reasons for the failure than just a missing directory.